File: test_gui.py
from playwright.sync_api import sync_playwright

with sync_playwright() as p:
    browser = p.chromium.launch(headless=True)
    page = browser.new_page()

    # ホームページに移動
    page.goto('http://localhost:3000')
    page.wait_for_load_state('networkidle')

    # スクリーンショットを撮影
    page.screenshot(path='/tmp/homepage.png', full_page=True)
    print("Screenshot saved to /tmp/homepage.png")

    # ページタイトルを確認
    title = page.locator('h1').text_content()
    print(f"Page title: {title}")

    # 機能カードを確認
    features = page.locator('[class*="Card"]').all()
    print(f"Found {len(features)} feature cards")

    # 新しいセッションページへ移動
    page.click('a:has-text("新しいセッション")')
    page.wait_for_load_state('networkidle')

    # セッション作成ページのスクリーンショット
    page.screenshot(path='/tmp/session_new.png', full_page=True)
    print("Screenshot saved to /tmp/session_new.png")

    # フォーム要素を確認
    client_name = page.locator('input[name="client_name"], #client_name')
    meeting_title = page.locator('input[name="meeting_title"], #meeting_title')

    print(f"Client name input visible: {client_name.is_visible()}")
    print(f"Meeting title input visible: {meeting_title.is_visible()}")

    browser.close()
    print("Test completed successfully!")

File: test_gui.mjs
import { chromium } from 'playwright';

async function test() {
  const browser = await chromium.launch({ headless: true });
  const page = await browser.newPage();

  // ホームページに移動
  await page.goto('http://localhost:3000');
  await page.waitForLoadState('networkidle');

  // スクリーンショットを撮影
  await page.screenshot({ path: '/tmp/homepage.png', fullPage: true });
  console.log("Screenshot saved to /tmp/homepage.png");

  // ページタイトルを確認
  const title = await page.locator('h1').textContent();
  console.log(`Page title: ${title}`);

  // 機能カードを確認
  const features = await page.locator('[class*="Card"]').all();
  console.log(`Found ${features.length} feature cards`);

  // 新しいセッションページへ移動
  await page.click('a:has-text("新しいセッション")');
  await page.waitForLoadState('networkidle');

  // セッション作成ページのスクリーンショット
  await page.screenshot({ path: '/tmp/session_new.png', fullPage: true });
  console.log("Screenshot saved to /tmp/session_new.png");

  // フォーム要素を確認
  const clientNameVisible = await page.locator('input[name="client_name"], #client_name').isVisible();
  const meetingTitleVisible = await page.locator('input[name="meeting_title"], #meeting_title').isVisible();

  console.log(`Client name input visible: ${clientNameVisible}`);
  console.log(`Meeting title input visible: ${meetingTitleVisible}`);

  await browser.close();
  console.log("Test completed successfully!");
}

test().catch(console.error);
